### PR TITLE
Improve readme

### PR DESCRIPTION
**Pull request by quietshu**
Feedback welcomed: https://github.com/zeit/swr/tree/improve-readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
-    onErrorRetry,
-
-    fetcher                                // default fetcher function
-  }
-)
+const { data, error, isValidating, revalidate } = useSWR(key, fetcher, options)
 ```
 
-#### `key` as a function
+Example usage:
```
Comment (pacocoursey) [inline README.md]: This seems out of place to me in the API section, can we move it down to the examples section under `### Local Configuration`?

File: README.md [hunk 2]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
+
+#### Return Values
+- `data`: data for the given key resolved by `fetcher` (or undefined if not loaded)  
+- `error`: error thrown by `fetcher` (or undefined)  
+- `isValidating`: if there's a request or revalidation loading  
+- `revalidate`: function to trigger the validation manually
+
+#### Options
+
+- `suspense = false`: enabled React Suspense mode [(details)](#suspense-mode)
```
Comment (pacocoursey) [inline README.md]:
```suggestion
- `suspense = false`: enable React Suspense mode [(details)](#suspense-mode)
```

File: README.md [hunk 3]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
+- `fetcher = undefined`: the default fetcher function
+- `revalidateOnFocus = true`: auto revalidate when window gets focused
+- `refreshInterval = 0`: polling interval (disabled by default)
+- `refreshWhenHidden = false`: polling when the window is invisible (if `refreshInterval` is enabled)
+- `shouldRetryOnError = true`: retry when fetcher has an error [(details)](#error-retries)
+- `dedupingInterval = 2000`: deduping requests with the same key
```
Comment (pacocoursey) [inline README.md]:
```suggestion
- `dedupingInterval = 2000`: dedupe requests with the same key in this time span
```

File: README.md [hunk 4]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
+```js
+const { data, error } = useSWR('/api/user', fetcher, { refreshInterval: 3000 })
+```
+which fetches data with `fetcher('/api/user')` every 3 seconds.
+
+#### Parameters
+
+- `key`: a unique key string for the request (or a function / null) [(advanced usage)](#conditional-fetching)  
+- `fetcher`: (_optional_) a Promise returning function to fetch your data [(details)](#data-fetching) 
+- `options`: (_optional_) options for this SWR hook
```
Comment (pacocoursey) [inline README.md]:
```suggestion
- `options`: (_optional_) an object of options for this SWR hook
```

File: README.md (hunk 1 above)
Comment (quietshu) [reply README.md]: Hmmm actually I think this can be removed because we have the introduction section already.

Comment (quietshu) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
-
-    fetcher                                // default fetcher function
-  }
-)
+const { data, error, isValidating, revalidate } = useSWR(key, fetcher, options)
 ```
 
-#### `key` as a function
+Example usage:
+```js
```
Comment (quietshu) [inline README.md]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
+const { data, error } = useSWR('/api/user', fetcher, { refreshInterval: 3000 })
```
Comment (quietshu) [inline README.md]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
+```
```
Comment (quietshu) [inline README.md]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
+which fetches data with `fetcher('/api/user')` every 3 seconds.
```
Comment (quietshu) [inline README.md]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -66,209 +64,282 @@ of `fetcher` and rerenders the component.
...
+
```
Comment (quietshu) [inline README.md]:
```suggestion
```

**Pull request merged**